### PR TITLE
Implement v2 for search filters wrapper with isolated icon filters and featured filters

### PR DESCRIPTION
**Pull request by nickgraz**
- Implement v2 for search filters wrapper with isolated icon filters and featured filters
- Will be used in the deliverables table in a future PR
- Story incoming in a future PR

### CODE REVIEW

**Comment by nickgraz** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2198?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#2198** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2198?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2197** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2197?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2198?utm_source=stack-comment).

File: src/components/common/SearchFiltersWrapperV2/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import IconFilters from './IconFilters';
+import FeaturedFilters from './FeaturedFilters';
+
+export type SearchInputProps = {
+  search: string;
+  onSearch: (search: string) => void;
+};
+
+export type FilterConfig = {
```
Comment (karthikbtf) [inline src/components/common/SearchFiltersWrapperV2/index.tsx:16]: should probably move this into IconFilters and let that be unaware of this file, there's a mutual dependence right now

**Review by karthikbtf — APPROVED** [review]
Nice 🙌  thanks for getting to this, you had mentioned wanting to do this sometime back.
One suggestion in there.

File: src/components/common/SearchFiltersWrapperV2/index.tsx (hunk 1 above)
Comment (nickgraz) [reply src/components/common/SearchFiltersWrapperV2/index.tsx:16]: It is shared intentionally, the idea though is that you can show the same filter config in two different ways, "featured" or within the filter icon popover. A common interface with two implementations

Comment (karthikbtf) [reply src/components/common/SearchFiltersWrapperV2/index.tsx:16]: Got it 👍, that makes sense.

**Pull request merged**